### DISCUSSION TITLE
How to suppress AWS profile info on right side of screen?

### DISCUSSION THREAD

**Discussion by joshuawhite929** [ohmyzsh/ohmyzsh#10726] [Help]
I'm currently using the agnoster theme with the aws plugin.  This plugin helps display your current aws profile information in the command prompt.  This is working well and I appreciate it.

What I don't like is that on the far right of the screen, I get the aws profile info displayed again as "<aws:someProfileName>"

If I set SHOW_AWS_PROMPT=false, the info in the command prompt and the info pinned at the far right both disappear.  Any idea how to remove just the aws profile info at the right of the screen?

Looking at this further, the aws plugin is setting the RPROMPT value ([see here](https://github.com/ohmyzsh/ohmyzsh/blob/master/plugins/aws/aws.plugin.zsh)):

```
# AWS prompt
function aws_prompt_info() {
  [[ -n "$AWS_PROFILE" ]] || return
  echo "${ZSH_THEME_AWS_PREFIX:=<aws:}${AWS_PROFILE:gs/%/%%}${ZSH_THEME_AWS_SUFFIX:=>}"
}

if [[ "$SHOW_AWS_PROMPT" != false && "$RPROMPT" != *'$(aws_prompt_info)'* ]]; then
  RPROMPT='$(aws_prompt_info)'"$RPROMPT"
fi
```

**Comment by mcornella** [comment]
Since they both use the same variable to check it's impossible to target one or the other. Since `$SHOW_AWS_PROMPT` is only read on plugin load by the `aws` plugin you could set it before OMZ is sourced and then unset it after OMZ is sourced. That way you'd see the prompt defined by the agnoster theme.

Another option is to override the `aws_prompt_info` with an empty body: 
```sh
aws_prompt_info() {}
```
or remove it from `$RPROMPT` manually:
```sh
RPROMPT="${RPROMPT//\$\(aws_prompt_info\)}"
```
